src/service/viewer-impl.js

```diff
@@ -321,60 +315,36 @@ export class Viewer {
             throw error;
           }) : null;
 
-    // Trusted viewer and referrer.
-    let trustedViewerResolved;
-    let trustedViewerPromise;
+    /** @private {!Promise<boolean>|undefined|boolean} */
```

Comment:
Hmm. To me this seems pretty confusing this type mixing. Is it helpful? Why not always keep it as Promise/undefined?

Reply:
Because this value is used as a boolean in L338. Couldn't think of a better way to handle this. Any suggestion?

Reply:
Refactored the code further bit. PTAL